transport_nantes/topicblog/views.py

```diff
@@ -1045,15 +1067,30 @@ def beacon_view(response, **kwargs):
 
     # We remove the ".gif" from the end of the token
     token = kwargs['token'][:-4]
-    email, send_record_id = token_valid(token)
-    if email and send_record_id:
-        # In case the token is valid, we update the open_time
-        send_record: SendRecordMarketingEmail
-        send_record = SendRecordMarketingEmail.objects.get(pk=send_record_id)
-        if not send_record.open_time:
-            send_record.open_time = datetime.now(timezone.utc)
-            send_record.save()
-            logger.info(f"{email} opened email (SR.id :{send_record.pk})")
+    send_record_class_string, send_record_id = token_valid(token)
+    if send_record_class_string:
+        send_record_class = \
+            apps.get_model("topicblog", send_record_class_string)
+        try:
+            if not send_record_id:
+                raise ValueError
+            # In case the token is valid, we update the open_time
+            send_record = send_record_class.objects.get(pk=send_record_id)
+        except ObjectDoesNotExist:
+            logger.info(f"No send record with id {send_record_id} in class "
+                        f"{send_record_class_string} has been found")
+        except ValueError:
+            logger.info(
+                f"The token {token} provided an incorrect "
+                f"ID : {send_record_id} "
+                f"(Class string : {send_record_class_string}")
+        else:
```

Comment:
What is this an `else` to?  It's indented too far to go with `if send_record_class_string`, but it's a bit weird to say `try...else`.  Also it's a bit weird to say `else if` on two lines.

Can you clean up and handle all the cases above?  What should `beacon_view()` do on those exceptions?  It currently returns a response no matter what, which I think is correct (serve the pixel, even if we learned nothing), but you should comment that it's what you mean.